File: backend/src/main/kotlin/com/project/advanced/project/domain/valueobject/DocumentoIdentidad.java
package com.project.advanced.project.domain.valueobject

public record DocumentoIdentidad(String numero: String) {
  public DocumentoIdentidad{
    if(numero == null || numero.isBlank() || numero.isEmpty){
      throw ReglaDominioException("El Documento Identidad no es valido")
    }

    numero = numero.trim()
  }
}

File: backend/src/main/kotlin/com/project/advanced/project/domain/valueobject/DocumentoIdentidad.kt
package com.project.advanced.project.domain.valueobject

import com.project.advanced.project.domain.exception.ReglaDominioException

data class DocumentoIdentidad(
    val numero: String,
) {
    init {
        if (numero.isBlank() || numero.isEmpty()) {
            throw ReglaDominioException("El Documento Identidad no es valido")
        }
    }
}
